plugins/gimp/file-jxl-load.cc

```diff
@@ -339,12 +349,22 @@ bool LoadJpegXlImage(const gchar *const filename, gint32 *const image_id) {
                       nullptr, pixels_buffer_2, GEGL_AUTO_ROWSTRIDE);
 
       g_clear_object(&buffer);
+      if (stop_processing) status = JXL_DEC_SUCCESS;
     } else if (status == JXL_DEC_SUCCESS) {
       // All decoding successfully finished.
       // It's not required to call JxlDecoderReleaseInput(dec.get())
       // since the decoder will be destroyed.
       break;
     } else if (status == JXL_DEC_NEED_MORE_INPUT) {
+      if (pixels_buffer_1 == nullptr) {
+        if (!SetJpegXlOutBuffer(&dec, &format, &buffer_size, &pixels_buffer_1))
```

Comment:
I wasn't sure how progressive encoding works in multi-frame images.

Reply:
We return JXL_DEC_NEED_IMAGE_OUT_BUFFER for every frame. That's why I think you also need to handle the DEC_FRAME events, otherwise you would have a memory leak by re-assigning pixels_buffer_1 without freeing it first.

Reply:
So that's why that leak happens. That was an existing issue, thanks: #638.

Reply:
Fixed by https://github.com/libjxl/libjxl/pull/1870/commits/2d4a1912dce442662d8ccb7afce93cad79e488f0. This commit results in 50% less peak memory consumption on loading `Wikipedia.jxl` for me.